app/components/upload_item_shared/index.test.tsx

```diff
@@ -0,0 +1,540 @@
+    let database: Database;
+
+    beforeEach(async () => {
+        jest.clearAllMocks();
+        database = (await TestHelper.setupServerDatabase(serverUrl)).database;
+        getFormattedFileSize.mockReturnValue('1024 KB');
+    });
+
+    const createMockFile = (overrides: Partial<UploadItemFile> = {}): UploadItemFile => ({
+        id: 'test-id',
+        clientId: 'test-client-id',
+        name: 'test-file.jpg',
+        extension: 'jpg',
+        size: 1024,
+        uri: 'file://test-uri',
+        failed: false,
+        width: 800,
+        height: 600,
+        mime_type: 'image/jpeg',
+        ...overrides,
+    });
+
+    describe('File Type Display Behavior', () => {
+        it('should display image thumbnail for image files', () => {
+            isImage.mockReturnValue(true);
+            const imageFile = createMockFile({
+                name: 'vacation.jpg',
+                mime_type: 'image/jpeg',
+            });
+
+            const {getByTestId, queryByText} = renderWithEverything(
+                <UploadItemShared
+                    file={imageFile}
+                    testID='test-upload'
+                />,
+                {database},
+            );
+
+            // Should show image component
+            expect(getByTestId('test-upload')).toBeTruthy();
+
+            // Should NOT show file info text for images
+            expect(queryByText('vacation.jpg')).toBeNull();
+            expect(queryByText('JPG')).toBeNull();
+        });
+
+        it('should display file info for document files', () => {
+            isImage.mockReturnValue(false);
+            const docFile = createMockFile({
+                name: 'report.pdf',
+                extension: 'pdf',
+                mime_type: 'application/pdf',
+            });
+
+            const {getByText} = renderWithEverything(
+                <UploadItemShared
+                    file={docFile}
+                    testID='test-upload'
+                />,
+                {database},
+            );
+
+            // Should show file info
+            expect(getByText('report.pdf')).toBeTruthy();
+            expect(getByText('PDF 1024 KB')).toBeTruthy();
+        });
+
+        it('should handle files without extension gracefully', () => {
+            isImage.mockReturnValue(false);
+            const fileWithoutExt = createMockFile({
+                name: 'document_no_extension',
+                extension: undefined,
+                mime_type: 'application/octet-stream',
+            });
+
+            const {getByText} = renderWithEverything(
+                <UploadItemShared
+                    file={fileWithoutExt}
+                    testID='test-upload'
+                />,
+                {database},
+            );
+
+            expect(getByText('document_no_extension')).toBeTruthy();
+            expect(getByText('DOCUMENT_NO_EXTENSION 1024 KB')).toBeTruthy(); // Shows extracted extension and size
```

Comment:
same here, should read 1Mb

I would also expect to see other tests that show B(ytes), Kb, Mb, Gb